js/src/hooks/useCountryCallingCodes.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { getCountries, getCountryCallingCode } from 'libphonenumber-js';
+
+/**
+ * Internal dependencies
+ */
+import useCountryKeyNameMap from './useCountryKeyNameMap';
+
+const toData = ( country, countryCallingCode, countryName ) => ( {
+	country,
+	countryCallingCode,
+	countryName,
+} );
+
+const toOption = ( country, countryCallingCode, countryName ) => ( {
+	key: country,
+	keywords: [ countryName, countryCallingCode, country ],
+	label: `${ countryName } (+${ countryCallingCode })`,
+} );
+
+export default function useCountryCallingCodes( as ) {
```

Comment:
:question: 
Do you envision some more formats, as it's a string? or could we make it `asOptions = false`, to make the available options/values more explicit.
Also, I have't found any usage of `useCountryCallingCodes( 'to-data' )` or `useCountryCallingCodes( anythingelse )`.
Do we even need that parameter?

Reply:
I created the hook with the general name `useCountryCallingCodes` and thought its data structure looks too particular if returns option structure directly. So I added the `as` parameter to indicate the returning data structure.

But considering it will probably only be used with this data structure, I renamed and simplified it by https://github.com/woocommerce/google-listings-and-ads/pull/917/commits/6e839d8b3b325fd85c5952d9bb0c1f48dc98887b